packages/@aws-cdk/pipelines/README.md

```diff
@@ -0,0 +1,542 @@
+
+    const pipeline = new CdkPipeline(this, 'Pipeline', {
+      // ...source and build information here (see below)
+    });
+
+    // Do this as many times as necessary with any account and region
+    // Account and region may different from the pipeline's.
+    pipeline.addApplicationStage(new MyApplication(this, 'Prod', {
+      env: {
+        account: '123456789012',
+        region: 'eu-west-1',
+      }
+    }));
+  }
+}
+```
+
+The pipeline is **self-mutating**, which means that if you add new
+application stages in the source code, or new stacks to `MyApplication`, the
+pipeline will automatically reconfigure itself to deploy those new stages and
+stacks.
+
+## CDK Versioning
+
+This library requires exactly CDK version `1.45.0`. The rest of your application must
+use the same version.
+
+It uses prerelease features of the CDK framework, which can be enabled by adding the
+following to `cdk.json`:
+
+```
+{
+  ...
+  "context": {
+    "@aws-cdk/core:newStyleStackSynthesis": true
+  }
+}
+```
+
+When bootstrapping, the environment variable `CDK_NEW_BOOTSTRAP=1` should be
+set (see the section called **CDK Bootstrapping**).
+
+## Defining the Pipeline (Source and Synth)
+
+The pipeline is defined by instantiating `CdkPipeline` in a Stack. This defines the
+source location for the pipeline as well as the build commands. For example, the following
+defines a pipeline whose source is stored in a GitHub repository, and uses NPM
+to build. The Pipeline will be provisioned in account `111111111111` and region
+`eu-west-1`:
+
+```ts
+class MyPipelineStack extends Stack {
+  constructor(scope: Construct, id: string, props?: StackProps) {
+    super(scope, id, props);
+
+    const sourceArtifact = new codepipeline.Artifact();
+    const cloudAssemblyArtifact = new codepipeline.Artifact();
+
+    const pipeline = new CdkPipeline(this, 'Pipeline', {
+      pipelineName: 'MyAppPipeline',
+      cloudAssemblyArtifact,
+
+      sourceAction: new codepipeline_actions.GitHubSourceAction({
+        actionName: 'GitHub',
+        output: sourceArtifact,
+        oauthToken: SecretValue.secretsManager('GITHUB_TOKEN_NAME'),
+        trigger: codepipeline_actions.GitHubTrigger.POLL,
+        // Replace these with your actual GitHub project name
+        owner: 'OWNER',
+        repo: 'REPO',
+      }),
+
+      synthAction: SimpleSynthAction.standardNpmSynth({
+        sourceArtifact,
+        cloudAssemblyArtifact,
+        projectName: 'MyAppPipeline-synth',
+
+        // Use this if you need a build step (if you're not using ts-node
+        // or if you have TypeScript Lambdas that need to be compiled).
+        buildCommand: 'npm run build',
+      }),
+    });
+  }
+}
+
+const app = new App();
+new MyPipelineStack(this, 'PipelineStack', {
+  env: {
+    account: '111111111111',
+    region: 'eu-west-1',
+  }
+});
+```
+
+You provision this pipeline by making sure the target environment has been
```

Comment:
Add a section header here “initial pipeline deployment” it’s an important section